openidcpy/client.py

```diff
@@ -0,0 +1,163 @@
+import base64
+import json
+import urllib
+import urlparse
+
+import requests
+from jwcrypto import jwt, jwk
+from datetime import datetime
+
+
+class OidcClient(object):
+  def __init__(self, discovery_uri, client_id, client_secret=None):
+    self.discovery_uri = discovery_uri
+    self.client_id = client_id
+    self.client_secret = client_secret
+
+  def _discover(self):
+    if not hasattr(self, 'well_known'):
+      self.well_known = self._get_url(self.discovery_uri)
+      self.certs = {item['kid']: item for item in
+                    self._get_url(get_property('jwks_uri', self.well_known))[
+                      'keys']}
+      self.auth_uri = get_property('authorization_endpoint',
+                                   self.well_known)
+      self.token_uri = get_property('token_endpoint', self.well_known)
+      self.logout_uri = get_property('end_session_endpoint',
+                                     self.well_known)
+
+  def _get_url(self, url):
+    resp = None
+    try:
+      resp = requests.get(url, verify=False)
+      if resp.status_code != 200:
+        raise CommunicationError(
+            'Could not connect to discovery endpoint, {}, Code: {}'.format(
+                self.discovery_uri, str(resp.status_code)))
+      return resp.json()
+    finally:
+      if resp is not None:
+        resp.close()
+
+  def get_auth_url(self, response_type, redirect_uri, scopes, state):
```

Comment:
Maybe call this create_auth_url(), as get implies that you are actually performing a GET.

Reply:
Done